include/seqan3/io/record.hpp

```diff
@@ -0,0 +1,347 @@
+ * \ingroup io
+ * \tparam field_types          The types of the fields available to the record in a meta::list.
+ * \tparam field_types_as_ids   A seqan3::fields type with seqan3::field IDs corresponding to field_types.
+ * \tparam selected_field_ids   A seqan3::fields type with the subset (and order) of the fields selected.
+ * \tparam tup_t                The tuple type being aggregated.
+ * \tparam field_no             The field we are currently processing.
+ *
+ * This metafunction "iterates" over the selected_field_ids and retrieves the corresponding typenames from
+ * field_types via their identifer in field_types_as_ids. It recursively builds up tup_t which it returns
+ * once the end of selected_field_ids is reached.
+ */
+template <typename field_types,
+          typename field_types_as_ids,
+          typename selected_field_ids,
+          typename tup_t = std::tuple<>,
+          size_t field_no = 0>
+struct generate_record_tuple_type
+{
+    //!\brief The return type.
+    using type = tup_t;
+};
+
+//!\cond
+template <typename field_types,
+          typename field_types_as_ids,
+          typename selected_field_ids,
+          typename tup_t,
+          size_t field_no>
+    requires field_no < selected_field_ids::as_array.size()
+struct generate_record_tuple_type<field_types, field_types_as_ids, selected_field_ids, tup_t, field_no>
+{
+    using new_tup_t = decltype(
+        std::tuple_cat(tup_t{},
+                       std::tuple<meta::at_c<field_types,
+                                             field_types_as_ids::index_of(selected_field_ids::as_array[field_no])>>{}));
+
+    using type = typename generate_record_tuple_type<field_types,
+                                                     field_types_as_ids,
+                                                     selected_field_ids,
+                                                     new_tup_t,
+                                                     field_no + 1>::type; // iterator over the selected_fields
+};
+//!\endcond
+
+} // namespace seqan3::detail
+
+namespace seqan3
+{
+
+// ----------------------------------------------------------------------------
+// record
+// ----------------------------------------------------------------------------
+
+/*!\brief The class template that file record are based on; behaves like an std::tuple.
+ * \ingroup io
+ * \tparam field_types          The types of the fields available to this record in a meta::list.
+ * \tparam field_types_as_ids   A seqan3::fields type with seqan3::field IDs corresponding to field_types.
+ * \tparam selected_field_ids   A seqan3::fields type with the subset (and order) of the fields selected.
+ *
+ * This class template behaves just like an std::tuple, with the exception that it provides an additional
+ * get<>() interface that can be addressed not by absolute position or actual type of the element,
+ * but also via a seqan3::field identifier.
```

Comment:
Contradiction: is addressing by _absolute position or actual type_ possible or not? 
Line 197 says No, Line 198 suggests Yes ("also"). 
=> I guess you wanted to say "not only", which is YES.

Reply:
Hm, I thought it was clear: the "additional interface" (a separate, new one) can be used via fields (and notthing else), but since it is an "additional interface" it is clear that it is independent of the other interfaces?

I will change the sentence in any case!

Reply:
Yes I understand your intent now, but it is not obvious in first reading. Please try to make this point more explicit.